src/windows.rs

```diff
@@ -472,7 +472,7 @@ pub fn lock(handle: FileDesc) -> Result<(), Error> {
 }
 
 /// Tries to lock a file but returns as soon as possible if already locked.
-pub fn try_lock(handle: FileDesc) -> Result<bool, Error> {
+pub fn try_lock(handle: FileDesc) -> Result<(), Error> {
```

Comment:
Can you return a more specific error here?

Something like:

```rust
pub enum TryLockError {
    LockFailed(Error),
    WouldBlock,
}
```

```suggestion
pub fn try_lock(handle: FileDesc) -> Result<(), TryLockError> {
```

The alternative is adding a method to `Error`  like `would_block(&self) -> bool`